### ISSUE TITLE
band! overdraw=true not working

### ISSUE THREAD

**Issue by bradcarman** [JuliaPlots/Makie.jl#1621]
How can I get the object from `band!()` to draw in front of a line?  The `band!()` function seems to always place the mesh at the lowest z-level regardless of the order when it's added or if `overdraw=true`

```julia
f = Figure()
a = Axis(f[1,1])
x = collect(0:0.1:100)
lines!(a, x, sin.(x))
band!(a, collect(25:75), ones(51), -ones(51), overdraw=true, color=:black)
```

![image](https://user-images.githubusercontent.com/40798837/151386661-6ad71439-8629-4781-950f-dac0830bf28b.png)

**Comment by bradcarman** [comment]
One final question, the transparency doesn't seem to be working...

```julia
f = Figure()
a = Axis(f[1,1])
x = collect(0:0.1:100)
lines!(a, x, sin.(x))
b=band!(a, collect(25:75), ones(51), -ones(51), overdraw=true, color=(:black,0.2), transparency=true)
translate!(b, 0, 0, 2) 
```

![image](https://user-images.githubusercontent.com/40798837/151394119-eb4381fc-ce37-4991-b637-0d0d68bcebd7.png)

Am I doing this correctly?

**Comment by bradcarman** [comment]
OK, I see now, I needed to do this in CairoMakie, now it works!

```julia
CairoMakie.activate!()
begin
    f = Figure()
    a = Axis(f[1,1])
    x = collect(0:0.1:100)
    lines!(a, x, sin.(x))
    b=band!(a, collect(25:75), ones(51), -ones(51), color=(:black,0.2))
    translate!(b, 0, 0, 2) 
    f    
end
```

![image](https://user-images.githubusercontent.com/40798837/151395316-a4821501-02c1-449f-a681-880e75a1978b.png)

**Comment by SimonDanisch** [comment]
I think transparency should be fixed in GLMakie: v0.5.0.. It does work for me at least...
It should also work on previous versions, but may be more finicky
